### PR TITLE
Feature/backend/crud devices api

### PR DESCRIPTION
**Pull request by sapiens-sapide**

### CODE REVIEW

File: src/backend/defs/go-objects/device.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+		close(getSet)
+		d.Locker.Unlock()
+	}(d.Locker, getSet)
+
+	return getSet
+}
+
+// IsValidDeviceType checks the `t` string against the constant `DeviceTypes`
+// returns true if `t` is a valid type for a device.
+func IsValidDeviceType(t string) bool {
```
Comment (gdchamal) [inline src/backend/defs/go-objects/device.go]: Is there not a way to define like an enumeration, instead of a delimited string ?

Comment (sapiens-sapide) [reply src/backend/defs/go-objects/device.go]: no `enum` type in GO. See https://github.com/golang/go/issues/19814 and the response from lance taylor : « A new language feature needs compelling use cases. All language features are useful, or nobody would propose them; the question is: are they useful enough to justify complicating the language and requiring everyone to learn the new concepts? What are the compelling use cases here? How will people use these? For example, would people expect to be able to iterate over the set of valid enum values, and if so how would they do that? Does this proposal do more than let you avoid adding default cases to some switches? »

Comment (gdchamal) [reply src/backend/defs/go-objects/device.go]: So forget enum name, think a map or a structure of valid values. Not a single string like this.

Comment (sapiens-sapide) [reply src/backend/defs/go-objects/device.go]: Could be a `map[string]bool`, but map are unordered ; Thus there is no way with a map to find the `default` string for device.  
Also, making it a simple string makes it easy to put it in a config file, easying edit of devices' types list.

Comment (sapiens-sapide) [reply src/backend/defs/go-objects/device.go]: Changed to an array in commit below

**Comment by gdchamal** [comment]
LGTM

**Pull request merged**